File: articles/mathematical-modeling-in-an-age-of-complexity/c/complexity_scenario_summary.c
#include <stdio.h>
#include <stdlib.h>

typedef struct {
    const char *key;
    const char *scenario_name;
    double stress_level;
    double interdependence_level;
    double uncertainty_level;
    double resilience_score;
    double equity_score;
    double adaptability_score;
} ComplexityScenario;

double fragility_score(ComplexityScenario s) {
    return 0.35 * s.stress_level +
           0.30 * s.interdependence_level +
           0.25 * s.uncertainty_level +
           0.10 * (1.0 - s.adaptability_score);
}

double robust_value(ComplexityScenario s) {
    double f = fragility_score(s);
    return 0.40 * s.resilience_score +
           0.30 * s.equity_score +
           0.30 * s.adaptability_score -
           0.20 * f;
}

int main(void) {
    ComplexityScenario scenarios[] = {
        {"baseline", "Baseline stress", 0.35, 0.45, 0.40, 0.72, 0.68, 0.65},
        {"compound_shock", "Compound shock", 0.78, 0.70, 0.72, 0.48, 0.52, 0.55},
        {"cascading_failure", "Cascading failure", 0.88, 0.86, 0.75, 0.32, 0.40, 0.42},
        {"adaptive_pathway", "Adaptive pathway", 0.65, 0.68, 0.70, 0.66, 0.70, 0.82}
    };

    printf("key,stress_level,interdependence_level,uncertainty_level,resilience_score,equity_score,adaptability_score,fragility_score,robust_value\n");

    for (int i = 0; i < 4; ++i) {
        printf("%s,%.6f,%.6f,%.6f,%.6f,%.6f,%.6f,%.6f,%.6f\n",
               scenarios[i].key,
               scenarios[i].stress_level,
               scenarios[i].interdependence_level,
               scenarios[i].uncertainty_level,
               scenarios[i].resilience_score,
               scenarios[i].equity_score,
               scenarios[i].adaptability_score,
               fragility_score(scenarios[i]),
               robust_value(scenarios[i]));
    }

    fprintf(stderr, "complexity_scenario_summary complete\n");
    return EXIT_SUCCESS;
}

File: articles/mathematical-modeling-in-an-age-of-complexity/fortran/complexity_scenario_summary.f90
program complexity_scenario_summary
  implicit none

  integer, parameter :: n = 4
  character(len=32) :: keys(n)
  real(8) :: stress(n), interdependence(n), uncertainty(n), resilience(n), equity(n), adaptability(n)
  real(8) :: fragility(n), robust(n)
  integer :: i

  keys = (/ 'baseline                      ', 'compound_shock                ', &
            'cascading_failure             ', 'adaptive_pathway              ' /)

  stress = (/ 0.35d0, 0.78d0, 0.88d0, 0.65d0 /)
  interdependence = (/ 0.45d0, 0.70d0, 0.86d0, 0.68d0 /)
  uncertainty = (/ 0.40d0, 0.72d0, 0.75d0, 0.70d0 /)
  resilience = (/ 0.72d0, 0.48d0, 0.32d0, 0.66d0 /)
  equity = (/ 0.68d0, 0.52d0, 0.40d0, 0.70d0 /)
  adaptability = (/ 0.65d0, 0.55d0, 0.42d0, 0.82d0 /)

  print '(A)', 'key stress interdependence uncertainty resilience equity adaptability fragility robust_value'

  do i = 1, n
    fragility(i) = fragility_score(stress(i), interdependence(i), uncertainty(i), adaptability(i))
    robust(i) = robust_value(resilience(i), equity(i), adaptability(i), fragility(i))
    print '(A,1X,F8.4,1X,F8.4,1X,F8.4,1X,F8.4,1X,F8.4,1X,F8.4,1X,F10.4,1X,F10.4)', &
      trim(keys(i)), stress(i), interdependence(i), uncertainty(i), resilience(i), equity(i), adaptability(i), fragility(i), robust(i)
  end do

contains

  real(8) function fragility_score(stress, interdependence, uncertainty, adaptability)
    implicit none
    real(8), intent(in) :: stress, interdependence, uncertainty, adaptability
    fragility_score = 0.35d0 * stress + 0.30d0 * interdependence + 0.25d0 * uncertainty + 0.10d0 * (1.0d0 - adaptability)
  end function fragility_score

  real(8) function robust_value(resilience, equity, adaptability, fragility)
    implicit none
    real(8), intent(in) :: resilience, equity, adaptability, fragility
    robust_value = 0.40d0 * resilience + 0.30d0 * equity + 0.30d0 * adaptability - 0.20d0 * fragility
  end function robust_value

end program complexity_scenario_summary
